### PR TITLE
DOMA-1138 create meters page

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**
This pr based on  #557 since changes in the scheme are needed here.

изображение
изображение
изображение
изображение

### CODE REVIEW

File: apps/condo/domains/meter/components/createMeterModal/MeterInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+import { ELECTRICITY_METER_RESOURCE_ID } from '../../constants/constants'
+import styled from '@emotion/styled'
+import { useValidations } from '@condo/domains/common/hooks/useValidations'
+
+const { Option } = Select
+
+const METER_INFO_INPUT_COL_SPAN = 11
+
+const getTariffNumberSelectOptions = () => {
+    return Array.from({ length: 4 }, (_, i) => i + 1)
```
Comment (SavelevMatthew) [inline apps/condo/domains/meter/components/createMeterModal/MeterInfo.tsx:15]: hard-coded 4?

File: apps/condo/domains/meter/components/createMeterModal/MeterInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
+import { Button, Col, DatePicker, Form, Input, Row, Select, Space, Typography } from 'antd'
```
Comment (SavelevMatthew) [inline apps/condo/domains/meter/components/createMeterModal/MeterInfo.tsx:1]: use DatePicker from common components

File: apps/condo/domains/meter/components/createMeterModal/MeterInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    const MoreParametersMessage = intl.formatMessage({ id: 'MoreParameters' })
+    const LessParametersMessage = intl.formatMessage({ id: 'LessParameters' })
+    const TariffsCountMessage = intl.formatMessage({ id: 'pages.condo.meter.TariffsNumber' })
+    const InstallationDateMessage = intl.formatMessage({ id: 'pages.condo.meter.InstallationDate' })
+    const CommissioningDateMessage = intl.formatMessage({ id: 'pages.condo.meter.CommissioningDate' })
+    const SealingDateMessage = intl.formatMessage({ id: 'pages.condo.meter.SealingDate' })
+    const VerificationDateMessage = intl.formatMessage({ id: 'pages.condo.meter.VerificationDate' })
+    const NextVerificationDateMessage = intl.formatMessage({ id: 'pages.condo.meter.NextVerificationDate' })
+
+    const validations = useValidations()
```
Comment (SavelevMatthew) [inline apps/condo/domains/meter/components/createMeterModal/MeterInfo.tsx:63]: see comment about validations

File: apps/condo/domains/meter/components/createMeterModal/ResourcesList.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import styled from '@emotion/styled'
+import { colors, shadows, transitions } from '@condo/domains/common/constants/style'
+import { IMeterResourceUIState } from '../../utils/clientSchema/MeterResource'
+
+
+const ResourceCardContainer = styled(FocusContainer)`
+  transition: ${transitions.allDefault};
+  
+    &:hover {
+      // TODO(nomerdvadcatpyat): change .hover to .elevated when pr #555 will be merged into master
```
Comment (SavelevMatthew) [inline apps/condo/domains/meter/components/createMeterModal/ResourcesList.tsx:14]: undone todo

File: apps/condo/domains/meter/components/createMeterModal/ResourcesList.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+const ResourceCard = ({ label, resourceId, onClick }: ResourceCardProps) => {
+    const Icon = resourceIdToIcon[resourceId]
+
+    return (
+        <ResourceCardContainer onClick={onClick}>
+            <Space>
+                <Icon />
+                <Typography.Text strong={true} style={{ fontSize: '16px' }}>{label}</Typography.Text>
```
Comment (SavelevMatthew) [inline apps/condo/domains/meter/components/createMeterModal/ResourcesList.tsx:37]: hard-coded fontStyle? May be `@condo/domains/common/constants/style` instead

File: apps/condo/domains/common/components/UnitInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
+import { useIntl } from '@core/next/intl'
```
Comment (pahaz) [inline apps/condo/domains/common/components/UnitInfo.tsx:1]: could you please move this component to the nearest domain? Maybe it's the property. The common domain is for non-domain logic and components

**Review by pahaz — APPROVED** [review]

File: apps/condo/pages/meter/create.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    return (
+        <>
+            <Head>
+                <title>{PageTitle}</title>
+            </Head>
+            <PageWrapper>
+                <PageContent>
+                    <Row gutter={[12, 40]}>
+                        <Col span={24}>
+                            <Typography.Title level={1} style={{ margin: 0 }}>{PageTitle}</Typography.Title>
```
Comment (Dimitreee) [inline apps/condo/pages/meter/create.tsx:31]: you sure u need this?
we got global style which should disable this margin

Screenshot 2021-09-14 at 13 54 06

File: apps/condo/domains/meter/utils/clientSchema/index.ts [hunk 8]
```diff
@@ -3,8 +3,44 @@
...
+
+export const resourceIds = [
+    COLD_WATER_METER_RESOURCE_ID,
+    HOT_WATER_METER_RESOURCE_ID,
+    GAS_SUPPLY_METER_RESOURCE_ID,
+    ELECTRICITY_METER_RESOURCE_ID,
+    HEAT_SUPPLY_METER_RESOURCE_ID,
+]
+
+export const resourceIdToIcon = {
```
Comment (Dimitreee) [inline apps/condo/domains/meter/utils/clientSchema/index.ts:32]:
```suggestion
resourceIdIconMap
```

File: apps/condo/domains/common/components/icons/MeterIcon.tsx [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import Icon from '@ant-design/icons'
+
+const MeterIconSVG = ({ width = 20, height = 20, bgColor = '#fff' }) => {
```
Comment (Dimitreee) [inline apps/condo/domains/common/components/icons/MeterIcon.tsx:4]: use color from style

File: apps/condo/domains/common/components/icons/AppealIcon.tsx [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import Icon from '@ant-design/icons'
+
+export const AppealIconSVG = ({ width = 20, height = 20, bgColor = '#fff' }) => {
```
Comment (Dimitreee) [inline apps/condo/domains/common/components/icons/AppealIcon.tsx:4]: use color from style

File: apps/condo/domains/meter/components/AccountNumberInput.tsx [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+    const NoAccountNumber = intl.formatMessage({ id: 'pages.condo.meter.NoAccountNumber' })
+    const AddAccountDescription = intl.formatMessage({ id: 'pages.condo.meter.AddAccountDescription' })
+    const AddAccountMessage = intl.formatMessage({ id: 'pages.condo.meter.AddAccount' })
+
+    const { CreateAccountModal, setIsCreateAccountModalVisible } = useCreateAccountModal()
+
+    return (
+        <Col span={24}>
+            <Row justify={'center'}>
+                <Col span={8} style={{ padding: '80px 0' }}>
```
Comment (Dimitreee) [inline apps/condo/domains/meter/components/AccountNumberInput.tsx:21]: css hack detected

File: apps/condo/domains/meter/components/MeterCard.tsx [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+
+    return (
+        <MeterCardWrapper>
+            <Row gutter={[10, 0]}>
+                <Col span={24}>
+                    <Row justify={'space-between'}>
+                        <Col>
+                            <Space>
+                                <Icon style={{ fontSize: '20px' }} />
+                                <Typography.Text style={{ fontSize: fontSizes.content }} strong={true}>
```
Comment (Dimitreee) [inline apps/condo/domains/meter/components/MeterCard.tsx:48]: you can use shorthand for boolean props

**Review by AntonAL — COMMENTED** [review]
image
On meter create page there is a mess of buttons.
Maybe to hide action bar, when there is no account for selected unit?

**Comment by AntonAL** [comment]
After trying to add wrong account number, modal panel disappears and input is displayed instead.
The same input jumps from modal to form item, it looks strange.
image

**Comment by nomerdvadcatpyat** [comment]
> image
> On meter create page there is a mess of buttons. Maybe to hide action bar, when there is no account for selected unit?

Yes, I agree that it will be better, made changes in 7797304

**Comment by nomerdvadcatpyat** [comment]
> After trying to add wrong account number, modal panel disappears and input is displayed instead.
> The same input jumps from modal to form item, it looks strange.
> image

This is done specifically so that after filling out the meter reading, you can make sure that the account number is correct and, if it is not correct, then change it

File: apps/condo/domains/meter/components/AccountNumberInput.tsx (hunk 11 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/meter/components/AccountNumberInput.tsx]: made it through antd grids

File: apps/condo/domains/common/hooks/useValidations.ts [hunk 13]
```diff
@@ -55,12 +57,18 @@ export const useValidations: () => ValidatorTypes = () => {
         }
     }
 
+    const numberValidator: Rule = {
+        pattern: /^\d+(\.?\d+)?$/g,
```
Comment (SavelevMatthew) [inline apps/condo/domains/common/hooks/useValidations.ts:61]: What about ","....Seems like some meters uses it as separator....so, it's not user friendly :)

File: apps/condo/domains/meter/components/MeterCard.tsx [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+import styled from '@emotion/styled'
+import { Alert, Col, Divider, Form, Input, Row, Space, Typography } from 'antd'
+import { resourceIdIconMap } from '../utils/clientSchema'
+import React, { useState } from 'react'
+import { BillingAccountMeterReading } from '../../../schema'
+import { IMeterFormState } from '../utils/clientSchema/Meter'
+import dayjs from 'dayjs'
+import { useIntl } from '@core/next/intl'
+import { fontSizes } from '@condo/domains/common/constants/style'
+import { useValidations } from '../../common/hooks/useValidations'
```
Comment (SavelevMatthew) [inline apps/condo/domains/meter/components/MeterCard.tsx:11]: `@condo`

File: apps/condo/domains/meter/constants/errors.js [hunk 15]
```diff
@@ -1,5 +1,7 @@
 const VALUE_LESS_THAN_PREVIOUS_ERROR = '[value:lessThanPrevious:'
+const EXISTING_METER_NUMBER = '[unique:alreadyExists:number]'
```
Comment (SavelevMatthew) [inline apps/condo/domains/meter/constants/errors.js:2]: meter? not intuitive

**Review by SavelevMatthew — APPROVED** [review]

File: apps/condo/domains/common/components/ActionBar.tsx [hunk 16]
```diff
@@ -4,9 +4,9 @@ import { Affix, Space } from 'antd'
 import React from 'react'
 import { css, jsx } from '@emotion/core'
 
-
 const actionBar = css`
     position: relative;
+    left: -24px;
```
Comment (dkoviazin) [inline apps/condo/domains/common/components/ActionBar.tsx:9]: No. You can look at Figma's screens

**Review by dkoviazin — CHANGES_REQUESTED** [review]

File: apps/condo/domains/common/components/ActionBar.tsx (hunk 16 above)
Comment (nomerdvadcatpyat) [reply apps/condo/domains/common/components/ActionBar.tsx:9]: On figma screens action bar look like this:
изображение
изображение

Now it looks like this:
изображение

Shouldn't it be shifted to the left?

**Review by dkoviazin — APPROVED** [review]

**Pull request merged**